src/docbuilder/queue.rs

```diff
@@ -77,32 +76,42 @@ impl DocBuilder {
         &mut self,
         builder: &mut RustwideBuilder,
     ) -> Result<bool> {
-        let conn = connect_db()?;
-
-        let query = conn.query(
-            "SELECT id, name, version
-                                     FROM queue
-                                     WHERE attempt < 5
-                                     ORDER BY priority ASC, attempt ASC, id ASC
-                                     LIMIT 1",
-            &[],
-        )?;
-
-        if query.is_empty() {
-            // nothing in the queue; bail
-            return Ok(false);
-        }
+        // This is in a nested scope to drop the connection before build_package is called,
+        // otherwise the borrow checker will complain.
+        let (id, name, version) = {
+            let conn = self.db.get()?;
+
+            let query = conn.query(
+                "SELECT id, name, version
+                                         FROM queue
+                                         WHERE attempt < 5
+                                         ORDER BY priority ASC, attempt ASC, id ASC
+                                         LIMIT 1",
+                &[],
+            )?;
+
+            if query.is_empty() {
+                // nothing in the queue; bail
+                return Ok(false);
+            }
+
+            let id: i32 = query.get(0).get(0);
+            let name: String = query.get(0).get(1);
+            let version: String = query.get(0).get(2);
 
-        let id: i32 = query.get(0).get(0);
-        let name: String = query.get(0).get(1);
-        let version: String = query.get(0).get(2);
+            (id, name, version)
```

Comment:
```suggestion
            let next_crate = query.get(0);
            (next_crate.get("id"), next_crate.get("name"), next_crate.get("version"))
```
This might need fiddling with the `let (id, name, version)` at the top for type inferrence.